pkg/jsonAPI/jsonapi.go

```diff
@@ -0,0 +1,514 @@
+					},
+					prevDays: PassRate{
+						Percentage:          prev.PassPercentage,
+						ProjectedPercentage: prev.PassPercentageWithKnownFailures,
+						Runs:                prev.Successes + prev.Failures,
+					},
+				},
+			}
+		} else {
+			newJobPassRate = PassRatesByJobName{
+				Name: v.Name,
+				Url:  v.TestGridUrl,
+				PassRates: map[string]PassRate{
+					latestDays: PassRate{
+						Percentage:          v.PassPercentage,
+						ProjectedPercentage: v.PassPercentageWithKnownFailures,
+						Runs:                v.Successes + v.Failures,
+					},
+				},
+			}
+		}
+
+		// FIXME: not sure if I need this
+		count := jobTestCount
+		// deleted additionalCount since we want the API to return everything
+		jobTests := report.ByJob[v.Name]
+		for _, test := range jobTests.TestResults {
+			if util.IgnoreTestRegex.MatchString(test.Name) {
+				continue
+			}
+			if count == 0 {
+				continue
+			}
+			count--
+
+			encodedTestName := url.QueryEscape(regexp.QuoteMeta(test.Name))
+
+			bugList := util.TestBugCache[test.Name]
+
+			failingTest := FailingTest{
+				Name: test.Name,
+				Url:  fmt.Sprintf("https://search.svc.ci.openshift.org/?context=1&type=bug&maxMatches=5&maxBytes=20971520&groupBy=job&search=%s", encodedTestName),
+				PassRate: PassRate{
+					Percentage: test.PassPercentage,
+					Runs:       test.Successes + test.Failures,
+				},
+				Bugs: bugList,
+			}
+
+			newJobPassRate.FailingTests = append(newJobPassRate.FailingTests, failingTest)
+		}
+
+		// NOTE - not sure if this needs to be declared as a pointer so future modifications are made by reference
+		passRatesSlice = append(passRatesSlice, newJobPassRate)
+
+	}
+
+	return passRatesSlice
+
+}
+
+// canaryTestFailures section
+func jsonCanaryTestFailures(result map[string]util.SortedAggregateTestResult) []CanaryTestFailInstance {
+	all := result["all"].TestResults
+
+	var canaryFailures []CanaryTestFailInstance
+
+	if len(all) <= 0 {
+		return nil
+	}
+
+	for i := len(all) - 1; i > len(all)-10; i-- {
+		test := all[i]
+		encodedTestName := url.QueryEscape(regexp.QuoteMeta(test.Name))
+		canaryFailures = append(canaryFailures,
+			CanaryTestFailInstance{
+				Name: test.Name,
+				Url:  fmt.Sprintf("https://search.svc.ci.openshift.org/?maxAge=168h&context=1&type=bug%%2Bjunit&name=&maxMatches=5&maxBytes=20971520&groupBy=job&search=%s", encodedTestName),
+				PassRate: PassRate{
+					Percentage: test.PassPercentage,
+					Runs:       test.Successes + test.Failures,
+				},
+			})
+	}
+	return canaryFailures
+}
+
+// job runs with failure groups
+func jsonFailureGroupList(report util.TestReport) []FailureGroup {
+
+	var failureGroups []FailureGroup
+	for _, fg := range report.FailureGroups {
+		failureGroups = append(failureGroups, FailureGroup{
+			Job:          fg.Job,
+			Url:          fg.Url,
+			TestFailures: fg.TestFailures,
+		})
+	}
+	return failureGroups
+}
+
+func jsonTestImpactingBugs(testImpactingBugs []util.Bug) []util.Bug {
+	return testImpactingBugs
```

Comment:
this is just returning its own arg?  that doesn't seem useful :)

Reply:
yeah. I just created a function because I wanted a function to correspond to every section in the dashboard. I guess I can just refer to it directly.